File: examples/java/l2-resources/EncryptionAtRest.java
package com.myorg;

import java.util.List;
import org.jetbrains.annotations.NotNull;
import software.constructs.Construct;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import org.mongodb.awscdk.resources.mongodbatlas.*;
import software.constructs.Node;


public class CdkJavaStack extends Stack {
    final static String PROJECT_ID = "PROJECT_ID";
    final static String PROFILE = "PROFILE";
    final static String ROLE_ID = "ROLE_ID";
    final static String MASTER_KEY_ID = "MASTER_KEY_ID";


    public CdkJavaStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        final String profile = getContextProp(this.getNode(), PROFILE);
        final String roleId = getContextProp(this.getNode(), ROLE_ID);
        final String projectId = getContextProp(this.getNode(), PROJECT_ID);
        final String customerMasterKeyId = getContextProp(this.getNode(), MASTER_KEY_ID);

        final AtlasEncryptionAtRest encryptionAtRest = AtlasEncryptionAtRest.Builder.create(this, "encryptionAtRest")
            .profile(profile)
            .projectId(projectId)
            .region(this.getRegion())
            .customerMasterKeyId(customerMasterKeyId)
            .roleId(roleId)
            .build();
    }

    public String getContextProp(final Node pNode, final String pPropName){
        @NotNull final Object obj = pNode.tryGetContext(pPropName);
        if (obj.toString().isEmpty()) {
            return "";
        }
        return obj.toString();
    }
}

File: examples/l1-resources/encryption-at-rest.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { CfnEncryptionAtRest } from 'awscdk-resources-mongodbatlas';

interface AtlasStackProps {
  readonly projId: string;
  readonly profile: string;
  readonly region: string;
  readonly customerMasterKeyId: string;
  readonly roleId: string;
  readonly requirePrivateNetworking: boolean;
}

export class CdkTestingStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const atlasProps = this.getContextProps();
    new CfnEncryptionAtRest(this, 'EncryptionAtRest', {
      projectId: atlasProps.projId,
      profile: atlasProps.profile,
      enabledForSearchNodes: true,
      awsKmsConfig: {
        enabled: true,
        region: atlasProps.region,
        customerMasterKeyId: atlasProps.customerMasterKeyId,
        roleId: atlasProps.roleId,
        requirePrivateNetworking: atlasProps.requirePrivateNetworking,
      },
    });
  }

  getContextProps(): AtlasStackProps {
    const projId = this.node.tryGetContext('projId');
    if (!projId) {
      throw "No context value specified for projId. Please specify via the cdk context."
    }

    const profile = this.node.tryGetContext('profile') ?? 'default';
    const customerMasterKeyId = this.node.tryGetContext('customerMasterKeyId');
    const region = this.node.tryGetContext('region');
    const roleId = this.node.tryGetContext('roleId');
    const requirePrivateNetworking = this.node.tryGetContext('requirePrivateNetworking') === 'true';

    return {
      projId,
      profile,
      region,
      customerMasterKeyId,
      roleId,
      requirePrivateNetworking,
    }
  }
}
